### PR TITLE
MapCube as_array function now respects masked maps

### PR DESCRIPTION
**Pull request by wafels**
The mapcube as_array function currently returns the image data from the mapcube maps as a single three dimensional numpy array.  This PR allows the as_array function to return a masked numpy array if at least one of the maps in the mapcube has a mask.  Otherwise, a normal numpy array is returned.

### CODE REVIEW

File: sunpy/map/mapcube.py [hunk 1]
```diff
@@ -317,14 +318,40 @@ def all_maps_same_shape(self):
         """
         return np.all([m.data.shape == self.maps[0].data.shape for m in self.maps])
 
+    def at_least_one_map_has_mask(self):
+        """
+        Tests if at least one map has a mask.
+        """
+        return np.any([m.mask is not None for m in self.maps])
+
+    def no_map_has_mask(self):
```
Comment (Cadair) [inline sunpy/map/mapcube.py]: is this the same as `not self.at_least_one_map_has_mask()`??

Comment (wafels) [inline sunpy/map/mapcube.py]: Yes, I believe you are right.  You just want to know if there is at least
one mask or not.  Thanks - this simplifies the code even more.

**Comment by Cadair** [comment]
:+1:

**Comment by ayshih** [comment]
:+1:

**Pull request merged**